doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md

```diff
@@ -1,66 +1,71 @@
-formula in a symbolic way in the input file (see
-{ref}`parameters:Initial_20temperature_20model`). An example
-of this latter case is this snippet of code discussed in
+## Syntax
+Input files have different ways of prescribing functional forms used by
+ASPECT. In many cases, the user can select parameters that feed into a
+fixed functional form defined inside an ASPECT plugin. In many others,
+ASPECT allows you to define your own function expression in a symbolic way
+in the input file (see {ref}`parameters:Initial_20temperature_20model`).
+An example of this latter case is this snippet of code discussed in
 {ref}`sec:cookbooks:convection_box_3d`:
 
 ```{literalinclude} ../../cookbooks/cookbooks/convection_box_3d/doc/initial.part.prm
 ```
 
-The formulas you can enter here need to use a syntax that is understood by the
-functions and classes that interpret what you write. Internally, this is done
-using the muparser library, see <http://muparser.beltoforion.de/>. The syntax
-is mostly self-explanatory in that it allows you to use the usual symbols `x`, `y`
-and `z` to reference coordinates (unless a particular plugin uses different
-variables, such as the depth), the symbol `t` for time in many situations, and
-allows you to use all of the typical mathematical functions such as sine and
-cosine. Based on the muparser library, deal.II supports additional functions,
-including `|` (the logical OR), `&` (the logical AND), `int()`, `ceil()`,
-`floor()`, `cot()`, `csc()`, `sec()`, `pow()`, `log()`, `erfc()`, `rand()`,
-and `rand_seed()`. For more detailed information, see
+These function expressions need to use a syntax that can be parsed by ASPECT.
+This parsing is done by the muparser library (<http://muparser.beltoforion.de/>).
+The syntax is mostly self-explanatory in that it allows you to use the usual
+symbols `x`, `y` and `z` for spatial coordinates
+(unless a particular plugin uses different variables, such as the depth),
+the symbol `t` for time in many situations, and allows you to use many common
+mathematical functions such as `||` (the logical OR), `&&` (the logical AND),
+`^` (raise x to the power of y), `sin()` and `cos()`
+(see <https://beltoforion.de/en/muparser/features.php> for a list of functions).
+Based on the muparser library, deal.II supports additional functions,
+including `int()`, `ceil()`, `floor()`, `cot()`, `csc()`, `sec()`,
+`pow()`, `erfc()`, `rand()`, and `rand_seed()`. For more detailed information, see
 <http://www.dealii.org/developer/doxygen/deal.II/classFunctionParser.html#details>.
 
-A common need for function expression is an if-else-statement, for example
-"if $1<x<4$ then output 1, else output 0." The muparser uses
+## Help with MuParser syntax
+If you would like to check that muparser is parsing your functions correctly,
+and to plot the values, you may consider running your expressions through
+`pymuparser` (<https://github.com/bobmyhill/pymuparser>),
```

Comment:
That's a very nice project! Would you want to also write a pull request for deal.II's documentation with similar information?

Reply:
Sure, I can do that. I was also wondering about extending the module with the additional deal.II functions. It wouldn't be hard to add some nesting logic, but this is probably not a top priority for the next couple of days.